Common/TableProducer/OccupancyTableProducer.cxx

```diff
@@ -0,0 +1,1159 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief  TPC PID - Calibration
+/// \author Rahul Verma (rahul.verma@iitb.ac.in) :: Marian I Ivanov (marian.ivanov@cern.ch)
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/DataModel/Multiplicity.h"
+
+#include "PWGLF/DataModel/LFStrangenessTables.h"
+
+#include "OccupancyTables.h"
+
+#include <typeinfo>
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/HistogramRegistry.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "DataFormatsFT0/Digit.h"
+#include "DataFormatsParameters/GRPLHCIFData.h"
+// #include "MetadataHelper.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+int32_t nBCsPerOrbit = o2::constants::lhc::LHCMaxBunches;
+
+struct occTableProducer {
+
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+
+  // declare production of tables
+  Produces<aod::OccIndexTable> GenOccIndexTable;
+  Produces<aod::Occs> GenOccTable;
+
+  Configurable<int> customOrbitOffset{"customOrbitOffset", 0, "customOrbitOffset for MC"};
+
+  // Histogram registry;
+  //  HistogramRegistry recoTracks  {"recoTracks"  , {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+  HistogramRegistry recoEvent{"recoEvent", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+
+  void init(InitContext const&)
+  {
+    LOGF(info, "Starting init");
```

Comment:
Please avoid unnecessary printouts at "info" level. Use "debug" level for debugging.

Reply:
I didn't got this. I have seen only LOG(info)<<"Comment" ;  or LOGF(info, "Comment"); Can You explain  a little what is meant by "debug" level and how is it implemented

Reply:
The logging macros are used as `LOG(level)`, where `level` can be `debug`, `fatal`, `error`, `warn` and `info`.

Reply:
I have implemented these changes